components/search_results_header/search_results_header.tsx

```diff
@@ -1,25 +1,24 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
 import React from 'react';
 import {Tooltip} from 'react-bootstrap';
 import {FormattedMessage} from 'react-intl';
 
 import OverlayTrigger from 'components/overlay_trigger';
 import Constants from 'utils/constants';
 
-export default class SearchResultsHeader extends React.PureComponent {
-    static propTypes = {
-        isExpanded: PropTypes.bool.isRequired,
-        children: PropTypes.node,
-        actions: PropTypes.shape({
-            closeRightHandSide: PropTypes.func.isRequired,
-            toggleRhsExpanded: PropTypes.func.isRequired,
-        }),
+type Props = {
+    isExpanded: boolean;
+    children?: React.ReactNode;
+    actions: {
+        closeRightHandSide: () => void;
+        toggleRhsExpanded: () => void;
     };
+};
 
-    render() {
+export default class SearchResultsHeader extends React.PureComponent<Props> {
```

Comment:
this can be converted to a functional component

```typescript
const SearchResultsHeader = (props: Props): React.ReactNode => { ... }
```

Or are we currently not converting those @hmhealey ?

Reply:
Yeah, we could. I thought there were more methods in this class, but I must be misremembering. I haven't actively been asking people to change that though.